.changeset/social-apes-dream.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/pg': patch
+---
+
+Make PostgreSQL observational-memory initialization atomic across processes and reject duplicate generation histories during schema upgrades.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)